script.js

```diff
@@ -1,4 +1,447 @@
+var L1GameMode = L1_PICK_GAME_MODE;
+var playerOneScore = 0;
+var playerTwoScore = 0;
+
+// ____________________________SUPPORTING FUNCTIONS_______________________________
+
+// 2 functions for player to select their order
+var selectAB = function () {
+  number = Number(`${arrayRolls[0]}` + `${arrayRolls[1]}`);
+  return number;
+};
+var selectBA = function () {
+  number = Number(`${arrayRolls[0]}` + `${arrayRolls[1]}`);
+  return number;
+};
+
+// standard dice roll functions
+var diceRoll = function () {
+  var randomNum = Math.floor(Math.random() * 6) + 1;
+  return randomNum;
+};
+
+// function to get the "scores" which is the sum of all the rolls
+// prepared code to be able to calculate for multiple players because the oder in which values are pushed into array is the same as order of player picking their numbers.
+
+var getScores = function (player) {
+  var myOutputValue = 0;
+  // input will help determine whose scores to calculate
+  var counter = player - 1;
+  while (counter < arrayForNums.length) {
+    myOutputValue = myOutputValue + arrayForNums[counter];
+    // this counter will help to only pull the numbers of the specific player from the array
+    counter += numOfPlayers;
+  }
+  return myOutputValue;
+};
+
+// function to generate the Leaderboard limited to 2 players for now
+var getLeaderboard = function () {
+  var myOutputValue = [
+    `Player2 : ${playerTwoScore}`,
+    ` Player1 : ${playerOneScore}`,
+  ];
+  if (playerOneScore >= playerTwoScore) {
+    myOutputValue = [
+      `Player1 : ${playerOneScore}`,
+      ` Player2 : ${playerTwoScore}`,
+    ];
+  }
+  return myOutputValue;
+};
+
+// function to refresh in order to go back to the main Choose Game mode
+var refresh = function () {
+  var myOutputValue = `Hello! Please choose the game mode you'll like to play. <br><br>Type in "Normal", "Lowest Win", "Auto", "Variable Dice", "Variable Player", "Knockout"`;
+  arrayRolls = [];
+  playerOneNum = 0;
+  playerTwoNum = 0;
+  arrayForNums = [];
+  currentPlayer = 1;
+  L1GameMode = L1_PICK_GAME_MODE;
+  return myOutputValue;
+};
+
+// _______________________ MAIN GAME _________________________________
+
+// this is the main function, where the L1 mode will determine which game to run
+
 var main = function (input) {
-  var myOutputValue = 'hello world';
+  var myOutputValue = "";
+  console.log("L1GameMode", L1GameMode);
+  if (input == "refresh") {
+    myOutputValue = refresh();
+  } else if (L1GameMode == L1_PICK_GAME_MODE) {
+    myOutputValue = startGame(input);
+  } else if (L1GameMode == L1_NORMAL_MODE) {
+    myOutputValue = normal(input);
+  } else if (L1GameMode == L1_LOWEST_MODE) {
+    myOutputValue = lowestWin(input);
+  } else if (L1GameMode == L1_AUTO_MODE) {
+    myOutputValue = auto(input);
+  } else if (L1GameMode == L1_VARIABLE_DICE_MODE) {
+    myOutputValue = variableDice(input);
+  } else if (L1GameMode == L1_VARIABLE_PLAYERS_MODE) {
+    myOutputValue = variablePlayers(input);
+  }
```

Comment:
Like how this was decomposed to the modes as well as the relevant output values. Well done 👍